src/Sulu/Bundle/ReferenceBundle/Domain/Model/Reference.php

```diff
@@ -184,13 +157,13 @@ public function setReferenceContext(string $referenceContext): static
     public function equals(ReferenceInterface $reference): bool
     {
         return
-            $this->resourceKey === $reference->resourceKey
-            && $this->resourceId === $reference->resourceId
-            && $this->referenceLocale === $reference->referenceLocale
-            && $this->referenceResourceKey === $reference->referenceResourceKey
-            && $this->referenceResourceId === $reference->referenceResourceId
-            && $this->referenceProperty === $reference->referenceProperty
-            && $this->referenceContext === $reference->referenceContext
-            && $this->referenceRouterAttributes === $reference->referenceRouterAttributes;
+            $this->getResourceKey() === $reference->getResourceKey()
+            && $this->getResourceId() === $reference->getResourceId()
+            && $this->getReferenceLocale() === $reference->getReferenceLocale()
+            && $this->getReferenceResourceKey() === $reference->getReferenceResourceKey()
+            && $this->getReferenceResourceId() === $reference->getReferenceResourceId()
+            && $this->getReferenceProperty() === $reference->getReferenceProperty()
+            && $this->getReferenceContext() === $reference->getReferenceContext()
+            && $this->getReferenceRouterAttributes() === $reference->getReferenceRouterAttributes();
```

Comment:
I don't see why change it but fine if we want to change it or not:


```suggestion
            $this->resourceKey === $reference->resourceKey
            && $this->resourceId === $reference->resourceId
            && $this->referenceLocale === $reference->referenceLocale
            && $this->referenceResourceKey === $reference->referenceResourceKey
            && $this->referenceResourceId === $reference->referenceResourceId
            && $this->referenceProperty === $reference->referenceProperty
            && $this->referenceContext === $reference->referenceContext
            && $this->referenceRouterAttributes === $reference->referenceRouterAttributes;
```